common/src/main/kotlin/entity/Interactions.kt

```diff
@@ -395,3 +399,50 @@ sealed class InteractionResponseType(val type: Int) {
 
     }
 }
+
+
+@KordPreview
+@Serializable
+data class DiscordApplicationCommandPermissions(
+        val id: String,
+        @SerialName("application_id")
+        val applicationId: String,
+        @SerialName("guild_id")
+        val guildId: String,
+        val permissions: List<DiscordApplicationCommandPermission>
+)
+
+@KordPreview
+@Serializable
+data class PartialDiscordApplicationCommandPermissions(
+        val id: String,
+        val permissions: List<DiscordApplicationCommandPermission>
+)
+
+@KordPreview
+@Serializable
+data class DiscordApplicationCommandPermission(
```

Comment:
This should be called `DiscordApplicationCommandPermissions`

Reply:
It only represents one permission `DiscordApplicationCommandPermissions` contains a list of `DiscordApplicationCommandPermission` since each instance of that entity which represents the perms for one role or user

Reply:
a GuildApplicationCommandPermissions contains ApplicationCommandPermissions
see: https://discord.com/developers/docs/interactions/slash-commands#guildapplicationcommandpermissions

Reply:
thanks for that  naming discord